crmsh/utils.py

```diff
@@ -2287,4 +2253,58 @@ def interface_choice():
     # should consider interface format like "ethx@xxx"
     interface_list = re.findall(r'(?:[0-9]+:) (.*?)(?=: |@.*?: )', out)
     return [nic for nic in interface_list if nic != "lo"]
+
+
+def interface_info(ipv6=False):
+    """
+    Get network information(IPv4 default) in local, include:
+        nic-ip dictionary
+        ip list
+        ip with mask list
+    IPv6 support
+    """
+    nic_ip_dict = {}
+    ip_list = []
+    ip_with_mask_list = []
+
+    cmd = "ip -{} -o addr show".format(6 if ipv6 else 4)
+    rc, out, err = get_stdout_stderr(cmd)
+    if rc != 0:
+        raise ValueError(err)
+
+    for line in out.split('\n'):
+        _, nic, _, ip_with_mask, *_ = line.split()
```

Comment:
Even though most probably it works, this line looks pretty weak. I would prefer to have a regular expression here. Anyway, if the tests pass with this solution I'm fine. Maybe add just a comment before this line with a an example like:
`2: eth0    inet 10.162.31.141/18 brd 10.162.63.255 scope global eth0\       valid_lft forever preferred_lft forever`

Reply:
I've added a comment here